pyproject.toml

```diff
@@ -235,8 +235,8 @@ dev = [
 
 # NOTE: those dependencies are used to test built in sources
 sources = [
-    "connectorx>=0.3.3 ; python_version >= '3.9'",
-    "connectorx>=0.4.0,<0.4.2 ; python_version >= '3.10'",
+    "connectorx>=0.4.2 ; python_version >= '3.9'",
```

Comment:
looks like support for Python 3.9 got dropped (`uv` does not resolve this dep on ci). my take: exclude Python 3.9. AFAIK we are not testing core source on it